### PR TITLE
Page transitions replaced by ZoomPageTransitionsBuilder

### PR DESCRIPTION
**Pull request by AlexV525**
The breaking change note for flutter/flutter#100812.

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (Hixie) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:11]: "excluding"

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (Hixie) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:23]: not sure what this sentence means

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (Hixie) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:31]: strike "that"

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (Hixie) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:25]: strike "as well"

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (Hixie) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:24]: what does "is not found" mean?

Comment (AlexV525) [reply src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:24]: That is https://github.com/flutter/flutter/pull/96375/commits/d5c96054ebb763b9374a74ec7ebc5840036645a6/packages/flutter/lib/src/material/page_transitions_theme.dart#L598-599

```dart
final PageTransitionsBuilder matchingBuilder =
    builders[platform] ?? const ZoomPageTransitionsBuilder();
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (AlexV525) [reply src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:23]: It's the same as L20-23, I think it might be bad to use the same sentence.

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (Hixie) [reply src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:24]: sorry to clarify when i ask a question like that what i really mean is "please update the text to answer the following question: ..." :-)

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (Hixie) [reply src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:23]: I don't think lines 22 and 23 here are grammatically correct English. I would suggest alternative wording but I'm not sure what they're trying to say so I'm not sure how to fix it.

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (Hixie) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:61]: is there a way to apply it to all platforms at once rather than having to list them all?

Comment (AlexV525) [reply src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:61]: Should we indicate that here?

**Comment by Hixie** [comment]
Learning more about the underlying PR, I think this doc needs to include the error message that happens when you find a Transform (or other widgets that we're introducing now), whereas previously you did not, and it should show how to fix it (e.g. using find.descendant).

**Comment by AlexV525** [comment]
I've added the tests migration and pointed out a few widgets related to this transition that might require migration.

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (domesticmouse) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:126]: I'm guessing this documentation should be held until this is in stable.

**Review by domesticmouse — COMMENTED** [review]
In general this LGTM, but I believe it should be held until after the next stable release.

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (AlexV525) [reply src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:126]: The PR should be held until the change is rolled. But shouldn't a breaking change note released along with the change? The *In stable release* should update after.

Comment (domesticmouse) [reply src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:126]: Ahh, yes you are right.

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
PTAL @sfshaza2

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:11]:
```suggestion
In order to ensure that libraries follow the latest OEM behavior,
the default page transition builders now use
`ZoomPageTransitionsBuilder` on all platforms (excluding iOS and macOS)
instead of `FadeUpwardsPageTransitionsBuilder`.
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:17]:
```suggestion
The `FadeUpwardsPageTransitionsBuilder` (provided with the first
Flutter release), defined a page transition that's
similar to the one provided by Android O. This page transitions builder
will eventually be deprecated on Android, as per Flutter's
[deprecation policy]({{site.url}}/resources/compatibility#deprecation-policy).
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:20]:
```suggestion
`ZoomPageTransitionsBuilder`, the new page transition builder for Android, Linux, and Windows,
defines a page transition that's similar to the one provided by Android Q and R.
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:28]:
```suggestion
`ZoomPageTransitionsBuilder` is used as the default.
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:90]:
```suggestion
If you used to try to find widgets but failed with *Too many elements*
using the new transition, and saw errors similar to the following:
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:107]:
```suggestion
Test before migration:
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:112]:
```suggestion
Test after migration:
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:120]:
```suggestion
Widgets that typically need to migrate the finder scope are:
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (sfshaza2) [reply src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:126]: Yes, @AlexV525 is correct.  :)

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Just a few grammatical edits and one clarification needed. thx!

**Comment by AlexV525** [comment]
Thanks @domesticmouse and @sfshaza2! The last step is to wait until the PR landed.

**Comment by AlexV525** [comment]
(as per https://github.com/flutter/flutter/pull/96375#issuecomment-1028342623)

**Comment by AlexV525** [comment]
The PR has been landed. Removing the WIP label and waiting if we have further reverts coming in.

**Comment by AlexV525** [comment]
ping @sfshaza2

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (AlexV525) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:127]:
```suggestion
Landed in version: v2.13.0-1.0.pre<br>
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (AlexV525) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:141]:
```suggestion
* [PR 100812][]
```

File: src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md
Comment (AlexV525) [inline src/release/breaking-changes/page-transition-replaced-by-ZoomPageTransitionBuilder.md:147]:
```suggestion
[PR 100812]: {{site.repo.flutter}}/pull/100812
```

**Review by domesticmouse — APPROVED** [review]

**Comment by AlexV525** [comment]
Another ping @sfshaza2

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**
